### PR TITLE
Fix scenrio tests

### PR DESCRIPTION
**Pull request by VivekVinushanth**
This PR fixes scenario tests

### CODE REVIEW

File: product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.1-sso-with-saml/src/test/java/org/wso2/identity/scenarios/sso/test/saml/SAMLSSOWithExternalAppTestCase.java [hunk 1]
```diff
@@ -21,48 +21,33 @@
...
-import org.wso2.identity.scenarios.commons.TestUserMode;
+import org.wso2.identity.scenarios.commons.*;
 import org.wso2.identity.scenarios.commons.clients.usermgt.remote.RemoteUserStoreManagerServiceClient;
 import org.wso2.identity.scenarios.commons.util.IdentityScenarioUtil;
 
 import java.util.HashMap;
 import java.util.Map;
 
-import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.*;
```
Comment (piraveena) [inline product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.1-sso-with-saml/src/test/java/org/wso2/identity/scenarios/sso/test/saml/SAMLSSOWithExternalAppTestCase.java:42]: remove `*` and use exact classes

File: product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.4-sso-with-oidc/4.1.4.3-basic-profile/src/test/resources/dcr-requests/dcr-request-2.json [hunk 2]
```diff
@@ -4,6 +4,6 @@
     "authorization_code"
   ],
   "redirect_uris": [
-    "https://sso-app-oidc-code-grant-2.com/callback"
+    "http://localhost"
```
Comment (piraveena) [inline product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.4-sso-with-oidc/4.1.4.3-basic-profile/src/test/resources/dcr-requests/dcr-request-2.json:7]: don't we need port here? if we don't give the port, to which port it is returning

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java [hunk 3]
```diff
@@ -89,7 +91,7 @@ public void testInit() throws Exception {
     @AfterClass(alwaysRun = true)
     public void atEnd() throws Exception {
         applicationManagementServiceClient = null;
-        userMgtClient.deleteRole(IDP_ROLE_1);
+//        userMgtClient.deleteRole(IDP_ROLE_1);
```
Comment (piraveena) [inline product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java:94]: why commented here

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java [hunk 4]
```diff
@@ -79,7 +81,7 @@ public void testInit() throws Exception {
         applicationManagementServiceClient = new ApplicationManagementServiceClient(sessionCookie, backendServiceURL,
                 configContext);
         userMgtClient = new UserManagementClient(backendServiceURL, sessionCookie);
-        userMgtClient.addRole(IDP_ROLE_1, new String[0], new String[]{PERMISSION_ADMIN_LOGIN});
+//        userMgtClient.addRole("Internal/"+IDP_ROLE_1, new String[0], new String[]{PERMISSION_ADMIN_LOGIN});
```
Comment (piraveena) [inline product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java:84]: why commented here

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java [hunk 5]
```diff
@@ -470,7 +472,7 @@ public void testUpdateRoles() {
 
             RoleMapping mapping = new RoleMapping();
             LocalRole localRole = new LocalRole();
-            localRole.setLocalRoleName(IDP_ROLE_1);
+            localRole.setLocalRoleName("Internal/admin");
```
Comment (piraveena) [inline product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java:475]: use constant for `"Internal/admin"`

File: product-scenarios/9-access-delegation/9.1-access-delegation-with-end-user-authorization/9.1.10-self-contained-access-token-as-authorization-grant/src/test/java/org/wso2/identity/scenarios/access/delegation/oauth2/selfcontainedtoken/OAuth2SelfContainedTokenImplicitGrantTest.java [hunk 6]
```diff
@@ -140,7 +140,7 @@ public void atEnd() throws Exception {
     public void intiAuthorizeRequest() throws Exception {
 
         HttpResponse response = oAuth2CommonClient
-                .sendAuthorizeGet(clientId, null, redirectUri, OAuth2Constants.ResponseTypes.TOKEN, null);
+                .sendAuthorizeGet(clientId, "openid profile", redirectUri, OAuth2Constants.ResponseTypes.TOKEN, null);
```
Comment (piraveena) [inline product-scenarios/9-access-delegation/9.1-access-delegation-with-end-user-authorization/9.1.10-self-contained-access-token-as-authorization-grant/src/test/java/org/wso2/identity/scenarios/access/delegation/oauth2/selfcontainedtoken/OAuth2SelfContainedTokenImplicitGrantTest.java:143]: use constant `"openid profile"`

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java (hunk 3 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java:94]: We are adding new role here. Re-using internal(admin role)

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java (hunk 4 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java:84]: We are adding new role here. Re-using internal(admin role)

Comment (piraveena) [reply product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java:84]: if this commented lines are not used, let's remove them

File: product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.4-sso-with-oidc/4.1.4.3-basic-profile/src/test/resources/dcr-requests/dcr-request-2.json (hunk 2 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.4-sso-with-oidc/4.1.4.3-basic-profile/src/test/resources/dcr-requests/dcr-request-2.json:7]: It just needs a valid host to obtain the callback.

**Comment by VivekVinushanth** [comment]
Sending  a new PR to update Cert: https://github.com/wso2/product-is/pull/14450/files

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/pom.xml [hunk 7]
```diff
@@ -146,10 +146,14 @@
             <artifactId>json-simple</artifactId>
             <scope>test</scope>
         </dependency>
+<!--        <dependency>-->
```
Comment (piraveena) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/pom.xml:149]: why it is commented

Comment (VivekVinushanth) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/pom.xml:149]: Was not needed. Will remove.

Comment (VivekVinushanth) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/pom.xml]: Fixed with https://github.com/wso2/product-is/pull/14317/commits/e509054962c4a6cdd77baec91ae9c4996af02854

File: product-scenarios/9-access-delegation/9.1-access-delegation-with-end-user-authorization/9.1.10-self-contained-access-token-as-authorization-grant/src/test/java/org/wso2/identity/scenarios/access/delegation/oauth2/selfcontainedtoken/OAuth2SelfContainedTokenImplicitGrantTest.java (hunk 6 above)
Comment (VivekVinushanth) [reply product-scenarios/9-access-delegation/9.1-access-delegation-with-end-user-authorization/9.1.10-self-contained-access-token-as-authorization-grant/src/test/java/org/wso2/identity/scenarios/access/delegation/oauth2/selfcontainedtoken/OAuth2SelfContainedTokenImplicitGrantTest.java]: Fixed with https://github.com/wso2/product-is/pull/14317/commits/1b4ab737ecfa0d7092d2c697a13b8ad39d9ad538

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java (hunk 5 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/14317/commits/1b4ab737ecfa0d7092d2c697a13b8ad39d9ad538

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java (hunk 4 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/14317/commits/1b4ab737ecfa0d7092d2c697a13b8ad39d9ad538

File: product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java (hunk 3 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.4-application-management/4.4.1-soap-apis/src/test/java/org/wso2/identity/scenarios/test/ApplicationManagementTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/14317/commits/1b4ab737ecfa0d7092d2c697a13b8ad39d9ad538

File: product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.1-sso-with-saml/src/test/java/org/wso2/identity/scenarios/sso/test/saml/SAMLSSOWithExternalAppTestCase.java (hunk 1 above)
Comment (VivekVinushanth) [reply product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.1-sso-with-saml/src/test/java/org/wso2/identity/scenarios/sso/test/saml/SAMLSSOWithExternalAppTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/14317/commits/1b4ab737ecfa0d7092d2c697a13b8ad39d9ad538

**Review by piraveena — APPROVED** [review]

**Pull request merged**